lib/miq_automation_engine/service_models/miq_ae_service_container_deployment.rb

```diff
@@ -0,0 +1,181 @@
+require 'net/scp'
+
+module MiqAeMethodService
+  class MiqAeServiceContainerDeployment < MiqAeServiceModelBase
+    expose :container_deployment_nodes, :association => true
+    expose :deployed_ems, :association => true
+    expose :deployed_on_ems, :association => true
+    expose :automation_task, :association => true
+    expose :roles_addresses
+    expose :container_nodes_by_role
+    expose :ssh_auth
+    expose :rhsm_auth
+
+    def assign_container_deployment_node(vm_id, role)
+      container_nodes_by_role(role).each do |deployment_node|
+        next unless deployment_node.vm_id.nil?
+        deployment_node.add_vm vm_id
+      end
+    end
+
+    def subscribe_deployment_master(deployment_master, user)
+      rhsub_user, rhsub_pass, rhsm_sku = rhsm_creds
+      subscribed = perform_agent_commands(deployment_master, user, ["sudo subscription-manager register --username='#{rhsub_user}'  --password='#{rhsub_pass}'"])[:stdout]
+      unless subscribed.include? "This system is already registered"
+        pool_id = perform_agent_commands(deployment_master, user, ["sudo subscription-manager list --available --matches=#{rhsm_sku} --pool-only"])[:stdout].split("\n").first.delete("\r")
+        perform_agent_commands(deployment_master, user, ["sudo subscription-manager attach --pool=#{pool_id}",
```

Comment:
can #{rhsm_sku} or #{pool_id} contain spaces?
I'd just quote all interpolations...

If any params can contain quotes, and/or come are user-controlled, we're in more trouble.